#!/usr/bin/env python
PACKAGE = "control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("speed_k1",   double_t,    0, "speed control param",    0.6, 0.0,   3.0)
gen.add("speed_k2",   double_t,    0, "speed control param",    0.6, 0.0,   3.0)
gen.add("k_front_distance",   double_t,    0, "speed control param",    1.0, 0.0,   5.0)
gen.add("k_behind_distance",   double_t,    0, "speed control param",    0.8, 0.0,   5.0)
gen.add("c_front_velocity",   double_t,    0, "speed control param",    0.8, 0.0,   5.0)
gen.add("c_behind_velocity",   double_t,    0, "speed control param",    0.6, 0.0,   5.0)
gen.add("k_speed_output",   double_t,    0, "speed control param",    60.0, 50.0,   400.0)
gen.add("k_brake_output",   double_t,    0, "speed control param",    2.5, 0.0,   10.0)
gen.add("min_preview_distance_for_search",   double_t,    0, "steer control param",    3.5, 0.0,   10.0)
gen.add("steer_ratio",   double_t,    0, "steer control param",    350.0, 100.0,   1200.0)
gen.add("heading_cost_factor",   double_t,    0, "steer control param",    1.0, 0.0,   5.0)
gen.add("straight_threshold",   double_t,    0, "steer control param",    0.01, 0.0,   0.2)
gen.add("k_curvature_factor",   double_t,    0, "speed control param",    5.0, 1.0,   10.0)
gen.add("dis_base", double_t, 0, "speed control param", 3.5, 0.0, 15.0)
gen.add("dis_speed_factor", double_t, 0, "speed control param", 0.25, 0.1, 1.5)
gen.add("dis_max", double_t, 0, "speed control param", 12.0, 5.0, 20.0)
gen.add("dis_min", double_t, 0, "speed control param", 4.0, 0.0, 10.0)
gen.add("preview_distance_base", double_t, 0, "steer control param", 3.55, 1.5, 6.5)
gen.add("preview_time", double_t, 0, "steer control param", 1.8, 1.0, 3.0)
gen.add("zero_brake_acc_threshold", double_t, 0, "speed control param", -0.5, -2.0, 0.0)
gen.add("zero_brake_speed_threshold", double_t, 0, "speed control param", 3.0, 0.0, 5.0)

gen.add("k_feedforward_acc",   double_t,    0, "speed control param",    0.75, 0.0,   10.0)
gen.add("k_acc_threshold",   double_t,    0, "speed control param",    0.25, 0.0,   10.0)
gen.add("k_acc_filter",   double_t,    0, "speed control param",    0.1, 0.0,   10.0)
gen.add("k_resmooth_threshold",   double_t,    0, "k_resmooth_threshold",   2.5, 0.0,   10.0)

exit(gen.generate(PACKAGE, "ky_node", "Param"))
